src/authz/mod.rs

```diff
@@ -0,0 +1,29 @@
+//! Types and traits for use in authorization. This module is only available if the `server` feature
+//! is enabled
+
+pub mod always;
+
+/// A trait that can be implemented on any type (such as a custom `User` or `Token` type) so that it
+/// can be authorized by an [`Authorizer`](Authorizer)
+pub trait Authorizable {
+    /// Returns the identity or username of the authenticated user
+    fn principal(&self) -> String;
+
+    /// Returns the groups the authenticated user is a member of, generally embedded on something
+    /// like a JWT or fetched from an upstream server
+    fn groups(&self) -> Vec<String>;
+}
+
+/// A trait for any system that can authorize any [`Authorizable`](Authorizable) type
+// TODO: Will this need to be async?
```

Comment:
Why would `authenticate` be async but `authorize` wouldn't?

Reply:
Authenticate for sure will need to contact something upstream to validate a token. Authorize should be checking something in memory. I was keeping this here until I implemented a real authorizer to see what I needed